src/measure_theory/set_integral.lean

```diff
@@ -257,8 +257,59 @@ hμ.integrable_at_filter hf.norm.is_bounded_under_le
 
 alias measure.finite_at_filter.integrable_at_filter_of_tendsto ← filter.tendsto.integrable_at_filter
 
-variables [measurable_space E] [borel_space E] [complete_space E] [second_countable_topology E]
-  [normed_space ℝ E]
+variables [measurable_space E]
+
+lemma integrable_add [opens_measurable_space E] {f g : α → E}
+  (h : univ ⊆ f ⁻¹' {0} ∪ g ⁻¹' {0})
+  (hf : measurable f) (hg : measurable g) :
+  integrable (f + g) μ ↔ integrable f μ ∧ integrable g μ :=
+begin
+  refine ⟨λ hfg, _, λ h, h.1.add hf hg h.2⟩,
+  rw [← indicator_add_eq_left h],
+  conv { congr, skip, rw [← indicator_add_eq_right h] },
+  rw [integrable_indicator_iff (hf (is_measurable_singleton 0)).compl],
+  rw [integrable_indicator_iff (hg (is_measurable_singleton 0)).compl],
+  exact ⟨hfg.integrable_on, hfg.integrable_on⟩
+end
+
+variables [borel_space E] [second_countable_topology E]
+/-- To prove something for an arbitrary measurable + integrable function in a second countable
+Borel normed group, it sufficient to show that
+* the property holds for (multiples of) characteristic functions;
+* is closed under addition;
+* the set of functions in the `L¹` space for which the property holds is closed.
+* the property is closed under the almost-everywhere equal relation. -/
+lemma integrable.induction {P : (α → E) → Prop}
+  (h_ind : ∀ (c : E) ⦃s⦄, is_measurable s → μ s < ⊤ → P (s.indicator (λ _, c)))
+  (h_sum : ∀ ⦃f g⦄, measurable f → measurable g → integrable f μ → integrable g μ → P f → P g → P (f + g))
```

Comment:
Can you again assume that the supports of the functions are disjoint here?

Reply:
yes